qemu/tests/cfg/device_assignment.cfg

```diff
@@ -7,7 +7,19 @@
     variants:
         - multi_pf_boot:
             type = boot
-            nics = "nic1 nic2"
+            nics = "nic1 nic2 nic3"
+            pci_assignable_nic1 = no
+            nic_model_nic1 = virtio
+            driver_option = 0
+            variants:
+                - 82576:
+                    pf_filter_re = 82576
+                - XL710:
+                    pf_filter_re = XL710
+                - 82599:
+                    pf_filter_re = 82599
+                - BCM57810:
+                    pf_filter_re = BCM57810
```

Comment:
Thanks all for your review. 
With this patch, testers don't need to configure "pf_filter_re" by manual and  test the cards one by one, which could save much effort. 
I agree with @luckyh , it may be annoying to developers to add variant for every new card.  
I don't know if there is a tradeoff between testers and developers. If you have any better idea, you are welcome to bring up. Thanks!